### PR TITLE
PMM-6396 add vmagent packaging into pmm-client

### PR DESCRIPTION
**Pull request by f41gh7**

### CODE REVIEW

**Review by EvgeniyPatlan — CHANGES_REQUESTED** [review]
Also it should be added into install_tarball script

**Comment by f41gh7** [comment]
> Also it should be added into install_tarball script

 As far as i understand, there is no need to add it to install_tarball, because install_tarball copies content of bin folder. May i ask you for an example?

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-1173-44f3523
client docker - perconalab/pmm-client-fb:PR-1173-44f3523
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-1173-44f3523.tar.gz

**Comment by JNKPercona** [comment]
Some Tests have Failed Please check: API: https://pmm.cd.percona.com/job/pmm2-api-tests/4596/ BATS: https://pmm.cd.percona.com/job/pmm2-testsuite/3978/ & UI: https://pmm.cd.percona.com/job/pmm2-ui-tests/5178/

**Review by EvgeniyPatlan — APPROVED** [review]
lgtm

**Review by BupycHuk — CHANGES_REQUESTED** [review]
Please, don't merge it for now

File: build/bin/build-client-source [hunk 1]
```diff
@@ -86,11 +86,31 @@ repack_toolkit() {
     "
 }
 
+prepare_vmagent_tarball() {
+    local repo_name=vmagent
+    local commit_hash=${VMAGENT_COMMIT:-vmagent_commit_hash}
```
Comment (atymchuk) [inline build/bin/build-client-source:91]:
```suggestion
    local commit_hash=${VMAGENT_COMMIT_HASH:-vmagent_commit_hash}
```
For naming consistency sake

**Review by atymchuk — CHANGES_REQUESTED** [review]

File: build/bin/build-client-source [hunk 2]
```diff
@@ -86,11 +86,31 @@ repack_toolkit() {
...
+    local commit_short=${commit_hash::6}
+    local tarball=${source_dir}/${repo_name}-${commit_short}.tar.gz
+    local repo_var=$(
+        echo "${repo_name}_commit" \
+            | sed -e 's:-:_:g' \
+            | tr '[[:lower:]]' '[[:upper:]]'
+    )
+    echo "${repo_var}=$commit_short" >> ${client_properties}
+
+   if [ -f "${tarball}" ]; then
```
Comment (AlekSi) [inline build/bin/build-client-source:101]: Three spaces for indentation there?

**Review by AlekSi — CHANGES_REQUESTED** [review]

**Review by atymchuk — DISMISSED** [review]

File: build/bin/build-client-source [hunk 3]
```diff
@@ -86,11 +86,31 @@ repack_toolkit() {
...
+    local tarball=${source_dir}/${repo_name}-${commit_short}.tar.gz
+    local repo_var=$(
+        echo "${repo_name}_commit" \
+            | sed -e 's:-:_:g' \
+            | tr '[[:lower:]]' '[[:upper:]]'
+    )
+    echo "${repo_var}=$commit_short" >> ${client_properties}
+
+    if [ -f "${tarball}" ]; then
+        echo ${tarball} already exists, skipping build
```
Comment (EvgeniyPatlan) [inline build/bin/build-client-source:102]: you are going to skip download not build

**Review by EvgeniyPatlan — CHANGES_REQUESTED** [review]

File: build/bin/build-client-source (hunk 3 above)
Comment (AlekSi) [reply build/bin/build-client-source]: Fixed it there, but it seems that "already exists, skipping build" is used incorrectly in other places too

**Review by atymchuk — APPROVED** [review]

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-1173-934c5f8
client docker - perconalab/pmm-client-fb:PR-1173-934c5f8
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-1173-934c5f8.tar.gz

**Review by BupycHuk — APPROVED** [review]

**Review by EvgeniyPatlan — APPROVED** [review]
LGTM

**Review by AlekSi — APPROVED** [review]

**Pull request merged**